src/test/util.ts

```diff
@@ -139,17 +139,26 @@ export function isRegExp(e: any): e is RegExp {
 
 export type Matcher = (value: string) => boolean;
 
+export type FilePatternFilter = {
+  re: RegExp;
+  line: number | null;
+}
+
 export function createMatcher(patterns: string | RegExp | (string | RegExp)[]): Matcher {
+  function cleanFilePattern(pattern: string): string {
+    if (pattern.includes(':'))
+      return cleanFilePattern(pattern.split(':')[0])
+    if (!pattern.startsWith('**/') && !pattern.startsWith('**/'))
+      return '**/' + pattern;
+    return pattern;
+  }
   const reList: RegExp[] = [];
   const filePatterns: string[] = [];
   for (const pattern of Array.isArray(patterns) ? patterns : [patterns]) {
     if (isRegExp(pattern)) {
       reList.push(pattern);
     } else {
-      if (!pattern.startsWith('**/') && !pattern.startsWith('**/'))
-        filePatterns.push('**/' + pattern);
-      else
-        filePatterns.push(pattern);
+      filePatterns.push(cleanFilePattern(pattern));
```

Comment:
Note that this also affects testIgnore and testMatch. Do we want the same logic there?

Reply:
I would say no, so reverted it.